src/System.Private.Interop/src/Internal/Runtime/CompilerHelpers/RuntimeInteropData.CoreRT.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            IntPtr openStub, closedStub, delegateCreationStub;
+            if (!TryGetMarshallersForDelegate(delegateTypeHandle, out openStub, out closedStub, out delegateCreationStub))
+            {
+                data = default(McgPInvokeDelegateData);
+                return false;
+            }
+
+            data = new global::System.Runtime.InteropServices.McgPInvokeDelegateData()
+            {
+                ReverseOpenStaticDelegateStub = openStub,
+                ReverseStub = closedStub,
+                ForwardDelegateCreationStub = delegateCreationStub
+            };
+            return true;
+        }
+        #region "Struct Data"
+        public override bool TryGetStructUnmarshalStub(RuntimeTypeHandle structureTypeHandle, out IntPtr unmarshalStub)
+        {
+            IntPtr marshalStub;
+            IntPtr destroyStub;
+            bool hasInvalidLayout;
+            int size;
+            return TryGetMarshallersForStruct(structureTypeHandle, out marshalStub, out unmarshalStub, out destroyStub, out hasInvalidLayout, out size);
+        }
+
+        public override bool TryGetStructMarshalStub(RuntimeTypeHandle structureTypeHandle, out IntPtr marshalStub)
+        {
+            IntPtr unmarshalStub;
+            IntPtr destroyStub;
+            bool hasInvalidLayout;
+            int size;
+            return TryGetMarshallersForStruct(structureTypeHandle, out marshalStub, out unmarshalStub, out destroyStub, out hasInvalidLayout, out size);
+        }
+
+        public override bool TryGetDestroyStructureStub(RuntimeTypeHandle structureTypeHandle, out IntPtr destroyStub, out bool hasInvalidLayout)
+        {
+            IntPtr marshalStub;
+            IntPtr unmarshalStub;
+            int size;
+            return TryGetMarshallersForStruct(structureTypeHandle, out marshalStub, out unmarshalStub, out destroyStub, out hasInvalidLayout, out size);
+        }
+
+        public override bool TryGetStructUnsafeStructSize(RuntimeTypeHandle structureTypeHandle, out int size)
+        {
+            IntPtr marshalStub;
+            IntPtr unmarshalStub;
+            IntPtr destroyStub;
+            bool hasInvalidLayout;
+            return TryGetMarshallersForStruct(structureTypeHandle, out marshalStub, out unmarshalStub, out destroyStub, out hasInvalidLayout, out size);
+        }
+
+        public override bool TryGetStructFieldOffset(RuntimeTypeHandle structureTypeHandle, string fieldName, out bool structExists, out uint offset)
+        {
+            ExternalReferencesTable externalReferences;
+            NativeParser entryParser;
+            structExists = false;
+            if (TryGetStructData(structureTypeHandle, out externalReferences, out entryParser))
+            {
+                structExists = true;
+                // skip the first 4 IntPtrs(3 stubs and size)
+                entryParser.GetUnsigned();
```

Comment:
There is SkipInteger method that you can use to here. It is slightly more efficient because of it does not compute the actual value.